pallets/payments/src/lib.rs

```diff
@@ -154,79 +154,61 @@ pub mod pallet {
 	/// interest. The storage map keys are the creator and the recipient, this
 	/// also ensures that for any (sender,recipient) combo, only a single
 	/// payment is active. The history of payment is not stored.
-	pub type Payment<T: Config> = StorageNMap<
+	pub type Payment<T: Config> = StorageDoubleMap<
 		_,
-		(
-			NMapKey<Blake2_128Concat, T::AccountId>,
-			NMapKey<Blake2_128Concat, T::AccountId>,
-			NMapKey<Blake2_128Concat, T::PaymentId>,
-		),
+		Blake2_128Concat,
+		// Sender
+		T::AccountId,
+		Twox64Concat,
```

Comment:
why not blake2 as well?

Reply:
Faster! and there's no way for users to influence the hash of the payment id as it is generated on-chain not from user input